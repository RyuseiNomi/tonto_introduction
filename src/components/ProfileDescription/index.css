.profile-description {
  padding: 0 10px 30px 20px;
  color: #E89A8A;
}

.name-and-sns {
  display: flex;
}

.sns {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.youtube-icon {
  color: red;
}

.profile-details p {
  white-space: pre-line;
  text-align: center;
}

/* ========= スマホサイズ_1 =========== */
@media (min-width: 280px) {
  .name-and-sns {
    justify-content: center;
    align-items: center;
  }

  .name-and-sns p {
    font-size: 24pt;
    padding-right: 6px;
  }

  .name-and-sns {
    flex-direction: column;
  }

  .profile-details p {
    font-size: 18px;
  }

  .sns img {
    display: flex;
    align-items: center;
    max-width: 50px;
    max-height: 50px;
    padding: 20px;
  }

  .youtube-icon {
    padding: 20px;
    font-size: 50px;
  }
}

/* ========= スマホサイズ_2 =========== */
@media (min-width: 576px) {
  .name-and-sns p {
    font-size: 30pt;
    padding-right: 8px;
  }

  .profile-details p {
    font-size: 20px;
  }

  .sns img {
    max-width: 60px;
    max-height: 60px;
    padding: 30px;
  }

  .youtube-icon {
    font-size: 60px;
    padding: 30px;
  }
}

/* ========= タブレットサイズ =========== */
@media (min-width: 768px) {
  .profile-description {
    padding: 50px 10px 50px 20px;
    text-align: left;
  }

  .name-and-sns {
    align-items: start;
    padding: 0 0 20px 0;
  }


  .name-and-sns p {
    font-size: 36pt;
    padding-right: 10px;
  }

  .sns img {
    max-width: 70px;
    max-height: 70px;
  }

  .youtube-icon {
    font-size: 70px;
  }
}

/* ========= PCサイズ =========== */
@media (min-width: 992px) {
  .profile-description {
    padding: 50px 10px 50px 20px;
  }

  .name-and-sns {
    padding: 0 0 20px 0;
  }

  .name-and-sns p {
    font-size: 36pt;
  }

  .sns img {
    display: flex;
    align-items: center;
    max-width: 50px;
    max-height: 50px;
  }

  .youtube-icon {
    font-size: 40pt;
  }
}
