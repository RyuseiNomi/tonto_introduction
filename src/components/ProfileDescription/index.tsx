import React from 'react';
import './index.css';
import { Row } from 'antd';
import { TwitterOutlined } from '@ant-design/icons';

type Props = {
  name: {
    jp: string;
    en: string;
  }
  twitter_url: string;
  description: {
    jp: string;
    en: string;
  }
  hobbies: {
    jp: string;
    en: string;
  }
  switch_code: string;
  discord_id: string;
  language: string;
}

function ProfileDescription(props: Props) {

  const { name, twitter_url, description, hobbies, switch_code, discord_id, language } = props;

  return (
    <div className="profile-description">
      <Row className="name-and-sns" justify="center">
        <p>{language === "jp" ? name.jp : name.en}</p>
        <a href={twitter_url} rel="noopener noreferrer" target="_blank"><TwitterOutlined /></a>
      </Row>
      <p className="profile-details">{language === "jp" ? description.jp : description.en}</p>
      <p className="profile-details">{language === "jp" ? hobbies.jp : hobbies.en}</p>
      <p className="profile-details">{switch_code}</p>
      <p className="profile-details">{discord_id}</p>
    </div>
  );
}

export default ProfileDescription;
