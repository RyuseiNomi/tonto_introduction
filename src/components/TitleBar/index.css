.title-bar {
  background-color: #E89A8A;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(25px + 2vmin);
  color: white;
  
  /* バックグランドイメージを設定するため */
  position: relative;
}

/* ========= スマホサイズ =========== */
@media (min-width: 280px) {
  .title-bar .background-image {
    /* 背景画像の相対位置を設定 */
    width: 120px;
    position: absolute;
    /* 左下に固定 */
    bottom: 0;
    left: 20px;
  }
}

/* ========= タブレットサイズ =========== */
@media (min-width: 576px) {
  .title-bar .background-image {
    /* 背景画像の相対位置を設定 */
    width: 120px;
    position: absolute;
    /* 左下に固定 */
    bottom: 0;
    left: 50px;
  }
}

/* ========= PCサイズ =========== */
@media (min-width: 768px) {
  .title-bar .background-image {
    /* 背景画像の相対位置を設定 */
    width: 130px;
    position: absolute;
    /* 左下に固定 */
    bottom: 0;
    left: 150px;
  }
}

/* スクロールを促すアイコン */
.scroll {
  padding-top: 50px;
  color: white;
}

.scroll span {
  position: absolute;
  left: 50%;
  width: 50px;
  height: 50px;
  margin-left: -30px;
  border-left: 3px solid white;
  border-bottom: 3px solid white;
  transform: rotate(-45deg);
  animation: go-scroll 2s infinite;
  box-sizing: border-box;
}

/* 矢印を下に動かすアニメーション */
@keyframes go-scroll {
  0% {
    transform: rotate(-45deg) translate(0, 0);
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    transform: rotate(-45deg) translate(-20px, 20px);
    opacity: 0;
  }
}
