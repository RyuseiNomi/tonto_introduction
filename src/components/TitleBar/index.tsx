import React from 'react';
import './index.css';
import { Radio } from 'antd';

type Props = {
  hook: Function;
  language: string;
}

function TitleBar(props: Props) {

  const { hook, language } = props;

  return (
    <div className="title-bar">
      <img className="title-image" src={language === "jp" ? `${process.env.PUBLIC_URL}/assets/title_jp.png` : `${process.env.PUBLIC_URL}/assets/title_en.png`} alt="background" />
      <img className="background-image" src={`${process.env.PUBLIC_URL}/hutaba_background.jpg`} alt="background" />
      <div className="language-selection">
        {/* TODO: 言語を切り替えられるようにする */}
        <Radio.Group value={language}>
          <Radio.Button value="jp" onClick={ () => hook("jp") }>日本語</Radio.Button>
          <Radio.Button value="en" onClick={ () => hook("en") }>English</Radio.Button>
        </Radio.Group>
      </div>
      {/* スクロールを促すアニメーション */}
      <p className="scroll"><span></span></p>
    </div>
  );
}

export default TitleBar;
