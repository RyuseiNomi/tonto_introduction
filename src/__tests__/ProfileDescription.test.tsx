import { cleanup } from '@testing-library/react';
import { render } from 'react-dom';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import ProfileDescription from './../components/ProfileDescription';
import TitleBar from './../components/TitleBar'; // 言語設定切り替えイベント実装のため

const sample_name = {jp: "テスト名前", en: "test-name"}
const sample_twitter_url = "sample-test-twitter-url"
const sample_description = {jp: "サンプル説明文", en: "sample-description"}
const sample_hobbies = {jp: "サンプル趣味", en: "sample-hobbies"}
const sample_sw_code = "0000-0000-0000-0000"

let container = null;
let root = null;

beforeEach(() => {
  // container の定義
  container = document.createElement("div");
  document.body.appendChild(container);
  root = createRoot(container);
});

afterEach(() => {
  // 定義した container の除去
  root.unmount(container);
  container.remove();
  container = null;
  cleanup;
});

// =============== snapshot test =================
it('自己紹介文が表示されていること', () => {
  const component = renderer.create(<ProfileDescription name={sample_name} twitter_url={sample_twitter_url} description={sample_description} hobbies={sample_hobbies} switch_code={sample_sw_code} language="jp" />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// =============== act tests =================
it('名前とプロフィール文が表示されていること', () => {
  const root = createRoot(container);
  act(() => {
    root.render(<ProfileDescription
        name={sample_name}
        twitter_url={sample_twitter_url}
        description={sample_description}
        hobbies={sample_hobbies}
        switch_code={sample_sw_code}
        language="jp"
      />);
  });
  expect(container.textContent).toBe("テスト名前サンプル説明文0000-0000-0000-0000名刺を表示");
});
