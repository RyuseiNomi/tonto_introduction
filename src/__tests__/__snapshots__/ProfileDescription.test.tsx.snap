// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`自己紹介文が表示されていること 1`] = `
<div
  className="profile-description"
>
  <div
    className="name-and-sns"
  >
    <p>
      テスト名前
    </p>
    <a
      href="sample-test-twitter-url"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="twitter-icon"
        src="assets/twitter_logo_blue.png"
      />
    </a>
  </div>
  <div
    className="profile-details"
  >
    <p>
      サンプル説明文
    </p>
    <p>
      0000-0000-0000-0000
    </p>
  </div>
  <div
    className="name-card-link"
  >
    <a
      className="name-card-button"
      href="/assets/name_card.jpg"
      rel="noopener noreferrer"
      target="_blank"
    >
      名刺を表示
    </a>
  </div>
</div>
`;
